### ISSUE TITLE
Parent constructor called with null

### ISSUE THREAD

**Issue by whitneyit** [Microsoft/TypeScript#3392]
The [Playground](http://www.typescriptlang.org/Playground#src=class%20A%20%7B%0A%09constructor(public%20str%3A%20string)%20%7B%0A%09%09%0A%09%7D%0A%7D%0Aclass%20B%20extends%20A%20%7B%0A%09constructor()%20%7B%0A%09%09super(null)%3B%0A%09%7D%0A%7D) doesn't complain when a parent class is called with null. Take the example below:

```ts
class A {
	constructor(public str: string) {
		
	}
}
class B extends A {
	constructor() {
		super(null);
	}
}
```

If I pass a `number` or `function` it will squawk but `null` seems fine.

Is this by design? I can't find any tests that say one way or another.

**Comment by RyanCavanaugh** [comment]
`null` is in the domain of all types.
